hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -974,3 +974,21 @@ def mock_finfo(dtype: DataType) -> FloatInfo:
         any=np.any,
         all=np.all,
     )
+
+
+def installed_array_modules() -> Dict[str, Any]:
+    """Returns list of array modules
+
+    A convenience wrapper for importlib.metadata.entry_points(). It has the
+    added benefit of working with both the original dict interface and the new
+    select interface, so this can be used warning-free in all modern Python
+    versions.
+    """
+    try:
+        eps = entry_points(group="array_api")
+    except TypeError:
+        # The select interface for entry_points was introduced in py3.10,
+        # supplanting its dict interface. We fallback to the dict interface so
+        # we can still find entry points in py3.8 and py3.9.
+        eps = entry_points()["array_api"]
+    return {ep.name: ep.load() for ep in eps}
```

Comment:
~~Instead of a comprehension, we'll want to unroll this as a for-loop so that we can catch-and-warn on any errors that `.load()` might raise.~~

Actually, let's just skip the `.load()` in this method; that lets us pass responsibility for error-handling onto the caller and annotate the returntype as `Dict[str, importlib.metadata.EntryPoint]`.